src/IECoreScene/ShaderNetworkAlgo.cpp

```diff
@@ -383,3 +385,238 @@ ShaderNetworkPtr ShaderNetworkAlgo::convertObjectVector( const ObjectVector *net
+	{
+		unduplicateEndPoints = true;
+		result.basis = SplineData::ValueType::Basis::linear();
+	}
+	else
+	{
+		result.basis = SplineData::ValueType::Basis::catmullRom();
+	}
+
+	const auto &positions = positionsData->readable();
+	const auto &values = valuesData->readable();
+
+	size_t n = std::min( positions.size(), values.size() );
+	for( size_t i = 0; i < n; ++i )
+	{
+		if( unduplicateEndPoints && ( i == 0 || i == n - 1 ) )
+		{
+			continue;
+		}
+
+		result.points.insert( typename SplineData::ValueType::Point( positions[i], values[i] ) );
+	}
+
+	return resultData;	
+}
+
+} // namespace
+
+IECore::ConstCompoundDataPtr ShaderNetworkAlgo::collapseSplineParameters( IECore::ConstCompoundDataPtr parameters )
+{
+	CompoundDataPtr newParametersData;
+
+	const auto &parms = parameters->readable();
+	for( const auto &maybeBasis : parameters->readable() )
+	{
+		if( !boost::ends_with( maybeBasis.first.string(), "Basis" ) )
+		{
+			continue;
+		}
+		const StringData *basis = runTimeCast<const StringData>( maybeBasis.second.get() );
+		if( !basis )
+		{
+			continue;
+		}
+
+
+		std::string prefix = maybeBasis.first.string().substr( 0, maybeBasis.first.string().size() - 5 );
+		IECore::InternedString positionsName = prefix + "Positions";
+		const auto positionsIter = parms.find( positionsName );
+		const FloatVectorData *floatPositions = nullptr;
+		const DoubleVectorData *doublePositions = nullptr;
+	
+		if( positionsIter != parms.end() )
+		{
+			floatPositions = runTimeCast<const FloatVectorData>( positionsIter->second.get() );
+			if( !floatPositions )
+			{
+				doublePositions = runTimeCast<const DoubleVectorData>( positionsIter->second.get() );
+			}
+		}
+
+		if( !( floatPositions || doublePositions ) )
+		{
+			continue;
+		}
+
+		IECore::InternedString valuesName = prefix + "Values";
+		const auto valuesIter = parms.find( valuesName );
+	
+		IECore::DataPtr foundSpline;
+		if( valuesIter != parms.end() )
+		{
+			if( floatPositions )
+			{
+				if( const FloatVectorData *floatValues = runTimeCast<const FloatVectorData>( valuesIter->second.get() ) )
+				{
+					foundSpline = loadSpline<SplineffData>( basis, floatPositions, floatValues );
+				}
+				else if( const Color3fVectorData *color3Values = runTimeCast<const Color3fVectorData>( valuesIter->second.get() ) )
+				{
+					foundSpline = loadSpline<SplinefColor3fData>( basis, floatPositions, color3Values );
+				}
+				else if( const Color4fVectorData *color4Values = runTimeCast<const Color4fVectorData>( valuesIter->second.get() ) )
+				{
+					foundSpline = loadSpline<SplinefColor4fData>( basis, floatPositions, color4Values );
+				}
+			}
+			else
+			{
+				if( const DoubleVectorData *doubleValues = runTimeCast<const DoubleVectorData>( valuesIter->second.get() ) )
+				{
+					foundSpline = loadSpline<SplineddData>( basis, doublePositions, doubleValues );
+				}
+			}
+		}
+
+		if( foundSpline )
+		{
+			if( !newParametersData )
+			{
+				newParametersData = parameters->copy();	
```

Comment:
Do we need the full copy here? Maybe we can make a new CompoundData that references the original parameter Data?

Reply:
Good spot, addressed.